src/lib/network_pool/transaction_pool.ml

```diff
@@ -327,6 +327,46 @@ struct
         t
     end
 
+    module Vk_refcount_table = struct
+      type t = (int * Verification_key_wire.t) Zkapp_basic.F_map.Table.t
+
+      let create () = Zkapp_basic.F_map.Table.create ()
+
+      let find t key = Zkapp_basic.F_map.Table.find t key
+
+      let inc ~key ~value t =
+        Zkapp_basic.F_map.Table.update t key ~f:(function
+          | None ->
+              (1, value)
+          | Some (x, value) ->
+              (x + 1, value) ) ;
+        Mina_metrics.(
+          Gauge.set Transaction_pool.vk_refcount_table_size
+            (Float.of_int (Zkapp_basic.F_map.Table.length t)))
+
+      let dec ~key ~value:_ t =
+        Zkapp_basic.F_map.Table.change t key ~f:(function
+          | None ->
+              None
+          | Some (x, value) ->
+              if x = 1 then None else Some (x - 1, value) ) ;
+        Mina_metrics.(
+          Gauge.set Transaction_pool.vk_refcount_table_size
+            (Float.of_int (Zkapp_basic.F_map.Table.length t)))
+
+      let lift_common t table_modify cmd =
+        User_command.extract_vks cmd
+        |> List.iter ~f:(fun vk -> table_modify ~key:vk.hash ~value:vk t)
+
+      let lift1 t table_modify cmd =
```

Comment:
Nit: `lift{1,2}` are not very good names, prefer `lift_hashed` and `lift_with_status`.